### PR TITLE
Add support of multiple scattering correction into AbsorptionCorrUtil

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

- Original Gitlab User Story: [PD290](https://code.ornl.gov/sns-hfir-scse/diffraction/powder/powder-diffraction/-/issues/290)
- This is  #32695 into  `main`

This PR modifies 
- `PDDetermineCharacterizations` to include some beamline specific log entries for wavelength log.
- existing `absorptioncorrutil.py` so that when a valid `ms_method` is supplied, the effective absorption correction factor (absorption and multiple scattering) is returned.

More specifically,
- when `ms_method` is set to "None", "", or None, `absorptioncorrutil.py` will only compute absorption correction with given `abs_method`.
- when `ms_method` is valid, `absorptioncorrutil.py` will compute both absorption correction with `abs_method` and multiple scattering correction with `ms_method`.  The effective absorption correction factor A_eff = A_abs/(1 - A_ms).
- the caching works on absorption correction only (`ms_method` disabled) as well as the effective absorption correction (`ms_method` valid).

**To test:**

- Testing requires access to the SNS netdrive (on analysis cluster)
- `ms_method` is not exposed to `SNSPowderReduction` yet, so it is not possible to testing the effective absorption correction with `SNSPowderReduction`.
- The standard call to `SNSPowderReduction` should still work.

```python
SNSPowderReduction("/SNS/PG3/IPTS-2767/nexus/PG3_46214.nxs.h5",
                   CalibrationFile="/SNS/PG3/shared/CALIBRATION/2020_1_11A_CAL/PG3_PAC_HR_d46168_2020_05_06.h5",
                   CharacterizationRunsFile="/SNS/PG3/shared/CALIBRATION/2020_2_11A_CAL/PG3_char_2020_01_04_PAC_limit_1.4MW.txt,/SNS/PG3/shared/CALIBRATION/2020_2_11A_CAL/PG3_char_2020_05_06-HighRes-PAC_1.4 MW.txt",
                   Binning=-0.001,
                   SaveAs="nexus",
                   TypeOfCorrection="SampleAndContainer",
                   SampleFormula="La-(B11)5.94-(B10)0.06",
                   MeasuredMassDensity=2.36,
                   ContainerShape="PAC06",
                   OutputFilePrefix='SC_absorption_',
                   OutputDirectory=".",
                   CacheDir=".",
                   )
```

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/mantid/utils/absorptioncorrutils.py [hunk 1]
```diff
@@ -48,13 +48,20 @@ def __get_instrument_name(wksp):
...
+    meta_wksp_name,
+    abs_method,
+    cache_dirs=[],
+    prefix_name="",
+    ms_method="",
+):
     """generate candidate cachefile names (full paths) and associated ascii hash
 
     :param meta_wksp_name: name of workspace contains relevant meta data for hashing
     :param abs_method: method used to perform the absorption calculation
```
Comment (peterfpeterson) [inline Framework/PythonInterface/mantid/utils/absorptioncorrutils.py:61]: add a note about this being required if the ms_method is supplied. Maybe it is better to supply that information on the ms_method parameter docs

File: Framework/PythonInterface/mantid/utils/absorptioncorrutils.py [hunk 2]
```diff
@@ -81,6 +88,7 @@ def __get_cache_name(meta_wksp_name, abs_method, cache_dirs=[], prefix_name=""):
                 "height": ws.run()['BL11A:CS:ITEMS:HeightInContainer'].lastValue(),
                 "sample_container": ws.run()['SampleContainer'].lastValue().replace(" ", ""),
                 "abs_method": abs_method,
+                "ms_method": ms_method,
```
Comment (peterfpeterson) [inline Framework/PythonInterface/mantid/utils/absorptioncorrutils.py:91]: Add a check above this point that the `abs_method` is supplied when `ms_method` is specified.

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Please add checks that absorption method is specified when multiple-scattering method is supplied. This should improve error messages when users violate the assumption that multiple-scattering requires absorption.

File: Framework/PythonInterface/mantid/utils/absorptioncorrutils.py (hunk 1 above)
Comment (KedoKudo) [reply Framework/PythonInterface/mantid/utils/absorptioncorrutils.py:61]: the docstring is updated as suggested

File: Framework/PythonInterface/mantid/utils/absorptioncorrutils.py (hunk 2 above)
Comment (KedoKudo) [reply Framework/PythonInterface/mantid/utils/absorptioncorrutils.py:91]: `abs_method` is a positional argument (i.e. mandatory by default) and `ms_method` is a keyword argument (optional with default to empty string).
There is no need to do any check here

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**